src/test/java/SalesforceModules/QuoteUtil.java

```diff
@@ -1037,28 +1053,208 @@ public static void changeStatusAsDraft_ss(String status, String logStep) throws
 
-        }catch (Exception e){}
+        } catch (Exception e) {
+        }
 
-        if(size==0){
+        if (size == 0) {
             KeywordUtil.takeScreenshotAndAttachInReport();
             System.out.println("no such status present");
             System.out.println(size);
-        }else {
+        } else {
             KeywordUtil.click(By.xpath(changeStatus_ss), "status changed as draft");
         }
 
         CasesUtil.clickSaveButton_ss("Saved");
     }
+
+    /**
+     * Create new quote combining all methods in billing section
+     *
+     * @param name,opportunityName the log
+     */
+    public static void createNewQuote_billing(String name, String opportunityName, String type) throws Exception {
+        clickQuoteButton_billing("Navigated to quote");
+        newButtonQuote_billing("Clicked new button");
+        checkPrimary("primary checkbox marked");
+        selectOpportunity_ss(opportunityName, opportunityName + " entered opportunities name");
+        selectAccount_billing(name, name + " entered account name");
+        selectType_ss(type, "selected type");
+        clickSaveButton("clicked save button");
+        addDiscountProduct_billing();
+    }
+
+
+    /**
+     * click Quote button for billing section
+     *
+     * @param logStep the log
+     */
+    public static void clickQuoteButton_billing(String logStep) throws InterruptedException {
+        KeywordUtil.delay(8000);
+        WebElement element = KeywordUtil.getDriver().findElement(QuoteObject.clickOnQuotes);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    /**
+     * click on new button for creating new Quote in billing section
+     *
+     * @param logStep the log
+     */
+    public static void newButtonQuote_billing(String logStep) throws InterruptedException {
+        KeywordUtil.delay(4000);
+        WebElement element = KeywordUtil.getDriver().findElement(QuoteObject.clickOnNew);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    /**
+     * Select the Account field for new quote in billing section
+     *
+     * @param logStep the log
+     * @param account the account name
+     */
+    public static void selectAccount_billing(String account, String logStep) throws InterruptedException {
+        boolean flag = false;
+        KeywordUtil.waitForVisible(ServiceSupportObject.searchAccounts_quote_ss);
+        KeywordUtil.click(ServiceSupportObject.searchAccounts_quote_ss, logStep);
+        KeywordUtil.inputText(ServiceSupportObject.searchAccounts_quote_ss, "TX", logStep);
+        KeywordUtil.delay(5000);
+        Thread.sleep(5000);
+        String accname = "(//lightning-base-combobox-formatted-text[contains(@title,'TX')])[1]";
```

Comment:
I can see that this xpath is not dynamic which we are making at runtime so it should be declared at the top as we do for other variables

Reply:
done